infra/terraform/terraform.go

```diff
@@ -81,6 +81,22 @@ func (r *terraform) Create(ctx context.Context, withInstaller bool) (installer i
 		return nil, trace.Errorf("No Terraform configs at %s", r.ScriptPath)
 	}
 
+	if r.Config.CloudProvider == constants.Azure {
+		err = r.azurePrepare(ctx)
```

Comment:
I think cloud provider should be an interface to be implemented for Azure, where this would become `Prepare`?